### PR TITLE
feat: Added Custom color option for text Font Color and background color using hex values 

### PR DESCRIPTION
**Pull request by Mukund-Tandon**
Fixes : #42 

Besides the default Colors added a option to give custom color by specifying the hex value of the color , User can also change the opacity of the color.

Before -

[Screencast from 06-04-23 03:46:47 PM IST.webm](https://user-images.githubusercontent.com/71614009/230354051-2eacd550-e070-43d1-8933-9bb035624fd1.webm)

After-

[Screencast from 06-04-23 04:11:21 PM IST.webm](https://user-images.githubusercontent.com/71614009/230354272-735d12b2-e898-406e-86b7-d84cdd4d34b9.webm)

### CODE REVIEW

File: lib/src/render/color_menu/color_picker.dart [hunk 1]
```diff
@@ -165,4 +193,115 @@ class _ColorPickerState extends State<ColorPicker> {
       ),
     );
   }
+
+  Widget _buildCustomColorItem(
+      _ColorType type,
+      TextEditingController colorController,
+      TextEditingController opacityController) {
```
Comment (LucasXu0) [inline lib/src/render/color_menu/color_picker.dart:200]:
```suggestion
      TextEditingController opacityController,
  ) {
```

**Comment by LucasXu0** [comment]
Hey, @Mukund-Tandon. Could you please consider adding tests to your code?

**Comment by Mukund-Tandon** [comment]
> Hey, @Mukund-Tandon. Could you please consider adding tests to your code?

Ok 👍,  I will add

**Comment by Mukund-Tandon** [comment]
Hey @LucasXu0, are there any more things needed to be done regarding this PR?

**Comment by LucasXu0** [comment]
Hey, @Mukund-Tandon. There're some files need to be formatted.

> Formatted lib/src/render/color_menu/color_picker.dart
Formatted test/render/color_menu/color_picker_test.dart
Formatted 212 files (2 changed) in 1.38 seconds.
Error: Process completed with exit code 1.

**Comment by Mukund-Tandon** [comment]
A new PR is Opened #74   by @Xazin  which fixes all issues of this PR

**Pull request closed without merging**